Acrolinx.Sidebar/ISidebar.cs

```diff
@@ -187,12 +189,67 @@ public bool SupportCheckSelection
 
     public class CheckedEventArgs : CheckEventArgs
     {
-        internal CheckedEventArgs(string checkId, Range range) : base(checkId)
+        internal CheckedEventArgs(string checkId, Range range, JArray embedCheckInformation, string inputFormat) : base(checkId)
         {
             Range = range;
+            EmbedCheckInformation = new JObject();
+            EmbedCheckInformation["embedCheckInformation"] = embedCheckInformation;
+            if (inputFormat.ToUpper().Equals("XML"))
```

Comment:
It might be good style to move that to an extra function. That function can be reused on other places too. It's also easier to test. That has the benefit that you can return the type directly without having a huge else if block.

    private static Format StringtoFormat(String inputFormat){
        if("xml".equals(inputFormat, OrdinalIgnoreCase))
        {
            return Format.Xml;
        }
        if("Text".equals(inputFormat, OrdinalIgnoreCase))
        {
            return Format.Text;
        }
        return originalCheckFormat;
   }

Reply:
@acrork Why does nobody loves the mighty switch statement? Just asking.
```
static Format convertToEnum(string inputFormat, Format originalCheckFormat)
{
  switch ((inputFormat ?? "").ToUpper())
  {
    case "XML":
      return Format.XML;
    case "HTML":
      return Format.HTML;
    case "MARKDOWN":
      return Format.Markdown;
    case "TEXT":
      return Format.Text;
    case "WORD_XML":
      return Format.Word_XML;
    default:
      return originalCheckFormat;
  }
}
```